app/tests/src/test_ingest_bem_pdfs.py

```diff
@@ -62,12 +100,60 @@ def test__ingest_bem_pdfs(caplog, app_config, db_session, policy_s3_file, file_l
         # assert "Temporary Assistance to Needy Families" not in second_chunk.content
         # assert "The Food Assistance Program" in second_chunk.content
         # assert math.isclose(second_chunk.mpnet_embedding[0], -0.82242084, rel_tol=1e-3)
-
-    # Clean up the temporary file
     if file_location == "local":
         os.remove("/app/tests/src/util/707.pdf.json")
 
 
+def test__enrich_text():
+    with smart_open("/app/tests/src/util/707.pdf", "rb") as file:
+        enriched_text_list = _enrich_texts(file)
+
+        assert len(enriched_text_list) == 45
+        first_enriched_text_item = enriched_text_list[0]
+        assert isinstance(first_enriched_text_item, EnrichedText)
+        assert first_enriched_text_item.headings == [Heading(title="Overview", level=1, pageno=1)]
+        assert first_enriched_text_item.type == "NarrativeText"
+        assert first_enriched_text_item.page_number == 1
+
+        other_enriched_text_item = enriched_text_list[13]
+        assert other_enriched_text_item.headings == [
+            Heading(title="Time and Attendance Review  Process", level=1, pageno=1),
+            Heading(title="Provider Errors", level=2, pageno=1),
+        ]
+        assert other_enriched_text_item.type == "NarrativeText"
+        assert other_enriched_text_item.page_number == 2
+
+
+def test__match_heading(mock_outline):
+    heading_with_extra_space = _match_heading(mock_outline, "Family Independence  Program (FIP)", 1)
+    assert heading_with_extra_space
+
+    heading_on_wrong_page = _match_heading(mock_outline, "Family Independence  Program (FIP)", 5)
+    assert heading_on_wrong_page is None
+
+
+def test__get_current_heading(mock_outline, mock_elements):
+    second_level_heading = _get_current_heading(
+        mock_outline,
+        mock_elements[1],
+        mock_outline[:2],
+    )
+    assert second_level_heading == [
+        Heading(title="Overview", level=1, pageno=1),
+        Heading(title="Family Independence Program (FIP)", level=2, pageno=1),
+    ]
+
+    replaced_second_level = _get_current_heading(mock_outline, mock_elements[2], mock_outline[:2])
+    assert replaced_second_level == [
+        Heading(title="Overview", level=1, pageno=1),
+        Heading(title="Program Goal", level=2, pageno=1),
+    ]
+
+    dropped_level = _get_current_heading(mock_outline, mock_elements[4], mock_outline)
+    assert dropped_level == [Heading(title="Test Level 1", level=1, pageno=2)]
```

Comment:
Is this `dropped_level` test intended to run [this line](https://github.com/navapbc/labs-decision-support-tool/pull/68/files#diff-d74557876d257737a4a1149fd26cd9a7ef610c44d243ebab30edc8036c65cdeeR144)? If so, it doesn't.

Reply:
See https://github.com/navapbc/labs-decision-support-tool/pull/68#discussion_r1742539215 for suggestion.